src/view/form-edit-view.js

```diff
@@ -160,6 +149,63 @@ class FormEditView extends AbstractView{
     evt.preventDefault();
     this._callback.rollUpClickHandler();
   }
+
+  #setInnerHandlers = () => {
+    this.element.querySelector('.event__type-group')
+      .addEventListener('change', this.#typeChangeHandler);
+    this.element.querySelector('.event__input--destination')
+      .addEventListener('change', this.#cityChangeHandler);
+  }
+
+  #typeChangeHandler = (evt) => {
+    if (evt.target.tagName === 'INPUT') {
+      const newType = evt.target.value;
+      this.updateData({
+        type: newType,
+        offers: (additionalOffers.find((el) => el.type === newType)).offers,
+      });
+    }
+  }
+
+  #cityChangeHandler = (evt) => {
+    const newCity = evt.target.value;
+    const isCityExist = CITY.includes(newCity);
+    evt.preventDefault();
+    if (newCity.length <= 0 || isCityExist === false) {
+      evt.target.setCustomValidity('please select a city from the list');
+    } else {
+      evt.target.setCustomValidity('');
+      this.updateData(
+        {
+          destination: {
+            description: generateDescription(DESCRIPTION_CITY),
+            name: newCity,
+            pictures: [
+              {
+                src: getRandomPhotos(),
+                description: generatePicDescription(PICTURES_CITY),
+              },
+            ],
+          },
+        },
+      );
+    }
+    evt.target.reportValidity();
+  }
+
+  reset = (point) => {
+    this.updateData(
+      FormEditView.parsePointToData(point),
+    );
+  }
+
+  static parsePointToData = (point) => Object.assign(point)
+
+
+  static parseDataToPoint = (data) => {
+    const point = {...data};
+    return point;
```

Comment:
```suggestion
  static parsePointToData = (point) => point


  static parseDataToPoint = (data) => data
```
сейчас это бесполезные операции. пусть не тратят просто так ресурсы